mapbox-map bound all events + markerclusters

mileszim opened the pull request:
I've bound all mapbox events, and added leaflet.markercluster to the project. MarkerCluster adds another bower dependency FYI.

binhums commented on addon/components/mapbox-map.js:
```diff
@@ -11,14 +11,50 @@ export default Ember.Component.extend({
-      });
 
+      // Bind Events
+      map.on('click', e => this.sendAction('onclick', e)); // Legacy
```
Can we send up the instance of the map instead of the click event?

binhums commented on addon/components/mapbox-map.js:
```diff
@@ -11,14 +11,50 @@ export default Ember.Component.extend({
 
-      if (this.get('center') && this.get('zoom')) {
+      // Setters
+      if (this.get('center')) {
```
Why have removed the requirement for `this.get('zoom')`

mileszim replied:
In the [mapbox.js docs](https://www.mapbox.com/mapbox.js/api/v2.3.0/l-map-class/#map-set-methods) for setView(), "zoom" is an optional parameter. This way you can center the map without also requiring a zoom param.

binhums replied:
Awesome :+1: Ideally we should add a test for this.

Have you tested it locally? Are you sure that the code works if you only specify center?

mileszim replied:
Tested it locally, works as expected.

binhums commented on addon/components/mapbox-map.js:
```diff
@@ -11,14 +11,50 @@ export default Ember.Component.extend({
+      map.on('locationfound', e => this.sendAction('locationfound', e));
+      map.on('locationerror', e => this.sendAction('locationerror', e));
+      map.on('popupopen', e => this.sendAction('popupopen', e));
+      map.on('popupclose', e => this.sendAction('popupclose', e));
```
So this changes the interface to `mapbox-map` to add any of these words as parameters, e.g.

```hbs
{{mapbox-map drag=(action... )}}
```

I think the Ember paradigm is to make these explicitly `onclick` instead of `click` to make it clear that it's an event we're talking about. E.g.

```hbs
{{mapbox-map ondrag=(action...)}}
```

What do you think?

binhums commented on addon/components/mapbox-map.js:
```diff
@@ -11,14 +11,50 @@ export default Ember.Component.extend({
-        this.sendAction('onclick', e);
-      });
 
+      // Bind Events
```
There's a bit of code duplication here. How would you feel about defining a constant like

```js
MAP_EVENTS = [
  'click',
  'dblclick',
  ...
];
```

and then saying something along the lines of

```js
MAP_EVENTS.forEach(event => {
  map.on(event), e => this.sendAction(event, e);
})
```

binhums commented:
Thanks for the PR @mileszim! A few comments about the eventing code.

The marker-cluster code looks great, if you want to separate that out into a separate PR I'm happy to merge that immediately.

binhums commented on tests/integration/components/mapbox-markercluster-test.js:
```diff
@@ -0,0 +1,24 @@
+    {{/mapbox-markercluster}}
+  `);
+
+  assert.equal(this.$().text().trim(), 'template block text');
```
This test is empty (and I think failing too). I'd either fill this test file out with some of the ways you expect the markercluster component to behave or just delete the file.

mileszim commented:
@binhums went ahead and addressed your comments. I deleted the test for the component, just FYI. Later this week or this weekend I will be adding a large suite of tests since we will be using this component heavily in a production app, so expect those coming your way in a week or two.

binhums commented:
Wonderful, thanks @mileszim. Can't wait to see these tests :+1:

I'll be publishing these as 0.3.1 but if you need a back port for 0.2.x just holler.

binhums commented:
Released and notes published 

https://github.com/binhums/ember-cli-mapbox/releases/tag/v0.3.1
